brainiak/reprsimil/brsa.py

```diff
@@ -169,24 +228,56 @@ class BRSA(BaseEstimator):
     beta_: array, shape=[conditions, voxels]
         The maximum a posterior estimation of the response amplitudes
         of each voxel to each task condition.
-    beta0_: array, shape=[n_nureg, voxels]
+    beta0_: array, shape=[n_nureg + n_base, voxels]
         The loading weights of each voxel for the shared time courses
-        not captured by the design matrix.
+        not captured by the design matrix. This helps capture the
+        structure of spatial covariance of task-unrelated signal.
+        n_base is the number of columns of the user-supplied nuisance
+        regressors plus one for DC component
+    X0_: array, shape=[time_points, n_nureg + n_base]
+        The estimated time course that is shared across voxels but
+        unrelated to the events of interest (design matrix).
+    beta0_null_: array, shape=[n_nureg + n_base, voxels]
+        The equivalent of beta0_ in a null model which does not
+        include the design matrix and response pattern beta
+    X0_null_: array, shape=[time_points, n_nureg + n_base]
+        The equivalent of X0_ in a null model which does not
+        include the design matrix and response pattern beta
 
     """
 
     def __init__(
-            self, n_iter=50, rank=None, GP_space=False, GP_inten=False,
-            tol=2e-3, auto_nuisance=True, n_nureg=6, verbose=False,
-            eta=0.0001, space_smooth_range=None, inten_smooth_range=None,
-            tau_range=10.0, init_iter=20, optimizer='BFGS', rand_seed=0):
+            self, n_iter=50, n_iter_inner=6, rank=None,
+            auto_nuisance=True, n_nureg=6, nureg_method='PCA',
+            DC_single=True,
+            GP_space=False, GP_inten=False,
+            space_smooth_range=None, inten_smooth_range=None,
+            tau_range=5.0, tau2_prior='invGamma',
+            eta=0.0001, init_iter=20, optimizer='BFGS',
+            rand_seed=0, anneal_speed=5,
+            tol=2e-3, verbose=False):
+
         self.n_iter = n_iter
+        self.n_iter_inner = n_iter_inner
         self.rank = rank
         self.GP_space = GP_space
         self.GP_inten = GP_inten
         self.tol = tol
         self.auto_nuisance = auto_nuisance
         self.n_nureg = n_nureg
+        if nureg_method == 'FA':
+            self.nureg_method = FactorAnalysis(n_components=n_nureg)
+        elif nureg_method == 'PCA':
+            self.nureg_method = PCA(n_components=n_nureg)
+        elif nureg_method == 'SPCA':
+            self.nureg_method = SparsePCA(n_components=n_nureg,
+                                          max_iter=20, tol=tol)
+        elif nureg_method == 'ICA':
+            self.nureg_method = FastICA(n_components=n_nureg, whiten=True)
+        else:
+            raise ValueError('nureg_method can only be FA, PCA, '
+                             'SPCA(for sparse PCA) or ICA')
```

Comment:
Thanks Mihai, I think I will make it able to take both. Part of is because some objects have optional parameters. And I do not want to force the user to look for scipy objects in order to use the method.